community/neo4j/src/test/java/schema/MultipleIndexPopulationStressIT.java

```diff
@@ -278,108 +302,118 @@ private void createRandomData( int count ) throws IOException
-        importer.doImport( new Input()
+        importer.doImport( new RandomDataInput( count ) );
+    }
+
+    protected Iterable<InputNode> randomNodes( int count )
+    {
+        return () -> new InputNodePrefetchingIterator( count );
+    }
+
+    private int randomPropertyValue( Random random )
+    {
+        return random.nextInt( 100 );
+    }
+
+    private class InputNodePrefetchingIterator extends PrefetchingIterator<InputNode>
+    {
+        private final int count;
+        private int i;
+
+        InputNodePrefetchingIterator( int count )
         {
-            @Override
-            public boolean specificRelationshipIds()
-            {
-                return false;
-            }
+            this.count = count;
+        }
 
-            @Override
-            public InputIterable<InputRelationship> relationships()
+        @Override
+        protected InputNode fetchNextOrNull()
+        {
+            if ( i >= count )
             {
-                return SimpleInputIteratorWrapper.wrap( "Empty", Collections.emptyList() );
+                return null;
             }
 
-            @Override
-            public InputIterable<InputNode> nodes()
+            try
             {
-                return SimpleInputIteratorWrapper.wrap( "Nodes", randomNodes( count ) );
+                return new InputNode( "Nodes", i, i, (long) i, randomProperties(), null, randomLabels(),
+                        null );
             }
-
-            @Override
-            public IdMapper idMapper()
+            finally
             {
-                return IdMappers.actual();
+                i++;
             }
+        }
 
-            @Override
-            public IdGenerator idGenerator()
-            {
-                return IdGenerators.fromInput();
-            }
+        private String[] randomLabels()
+        {
+            return random.randoms().selection( TOKENS, 1, TOKENS.length, false );
+        }
 
-            @Override
-            public Collector badCollector()
+        private Object[] randomProperties()
+        {
+            String[] keys = random.randoms().selection( TOKENS, 1, TOKENS.length, false );
+            Object[] result = new Object[keys.length * 2];
+            int i = 0;
+            for ( String key : keys )
             {
-                try
-                {
-                    return new BadCollector( fs.openAsOutputStream(
-                            new File( directory.directory(), "bad" ), false ), 0, 0 );
-                }
-                catch ( IOException e )
-                {
-                    throw new RuntimeException( e );
-                }
+                result[i++] = key;
+                result[i++] = randomPropertyValue( random.random() );
             }
-        } );
+            return result;
+        }
     }
 
-    protected Iterable<InputNode> randomNodes( int count )
+    private class RandomDataInput implements Input
```

Comment:
If you can, try to keep random refactorings separate from the actual changes, produces cleaner PRs